### ISSUE TITLE
Sporadic cosmetic bug with freezeColumns - partially hidden first non-frozen column

### ISSUE THREAD

**Issue by reczy** [glideapps/glide-data-grid#225]
Hello,

I'd like to report a minor cosmetic bug with freezeColumns. I saw #171 and #91 but this looks different.

To reproduce:

1. Visit the [Freeze Columns](https://glideapps.github.io/glide-data-grid/?path=/story/glide-data-grid-dataeditor-demos--freeze-columns) story
2. With the grid in its non-frozen state (ie, starting state when you visit/refresh the page), change the width of the browser window so that the header word "Title" is barely fully visible (see image below)
3. Focus on the "First name" of the first row, use your right arrow key to move right until you get to the "Title" column
4. Use your left arrow key to move back to the frozen "First name" column.

Following these steps, you will notice that even though the first column is in focus, the grid hasn't adjusted the non-frozen columns to line up evenly with the frozen column (meaning, the "Last name" column is partially hidden). Moving right once more to the "Last name" column with the right arrow key solves the problem.

You will note that if you follow the above steps but instead move right to the "More info" column (step 3), there is no issue.

I am using: 
Chrome: Version 99.0.4844.83 (Official Build) (arm64)
MacOS 12.3 (M1)

# Starting Point: 
![CleanShot 2022-03-26 at 12 04 46](https://user-images.githubusercontent.com/1100860/160249929-ba4b1821-0e83-4c03-968b-495a98032064.png)

# How it looks after following the steps above
![CleanShot 2022-03-26 at 12 05 33](https://user-images.githubusercontent.com/1100860/160249956-a5df9858-413f-4f2b-8d76-9a364976c149.png)

# In action
![CleanShot 2022-03-26 at 12 09 21](https://user-images.githubusercontent.com/1100860/160250091-b5b711c8-4bc7-4b9f-a71e-38e310a028d8.gif)

**Comment by jassmith** [comment]
Thank you for the report, will fix.

**Comment by jassmith** [comment]
Can you please try and see if you can reproduce this on the 4.0.0 branch? I put in some fixes for calculation of cell locations.

**Comment by jassmith** [comment]
I have not yet been able to reproduce the issue, are you able to reproduce on firefox?

**Comment by jassmith** [comment]
Okay I managed to reproduce the issue on firefox, I believe this is already fixed in 4.0.0. Please re-open if you are able to repro on the 4.0.0 branch.

**Comment by reczy** [comment]
Great, thanks. I was able to reproduce in firefox and safari. 

I'll let you know if it shows up after you release 4.0.0. Many thanks for the prompt response and consideration.

**Comment by jassmith** [comment]
I've just published 4.0.0-alpha1 on npm if you wish to test

**Comment by reczy** [comment]
Thank you. I tried running storybook locally on the 4.0.0 branch but ran into some errors (I'm guessing related to my own setup). 

I'll look to try out the new alpha in my own code when I have a chance and report back.

**Comment by jassmith** [comment]
Would you mind telling me what errors you ran into so I can make sure there
are no configuration issues I need to resolve?

**Comment by reczy** [comment]
Sure. Haven't run storybook locally before so I wouldn't be surprised if it's something silly/dependency-related on my end.

Clone glide-data-grid and pull 4.0.0 branch.

Run `npm install`

Run `npm run storybook`


There I see a bunch of errors like this 

```
ERROR in ./packages/cells/src/cell.stories.tsx
Module not found: Error: Can't resolve '@glideapps/glide-data-grid' in '/Users/mike/dev/opensource/glide-data-grid/packages/cells/src'
 @ ./packages/cells/src/cell.stories.tsx 22:0-91 59:31-50 109:25-35 124:16-28 137:16-28 151:16-28 166:16-28 180:16-28 195:16-28 209:16-28
 @ . sync ^\.(?:(?:^|\/|(?:(?:(?!(?:^|\/)\.).)*?)\/)(?!\.)(?=.)[^/]*?\.stories\.tsx)$
 @ ./generated-stories-entry.js
 @ multi ./node_modules/@storybook/core-client/dist/esm/globals/polyfills.js ./node_modules/@storybook/core-client/dist/esm/globals/globals.js (webpack)-hot-middleware/client.js?reload=true&quiet=false&noInfo=undefined ./storybook-init-framework-entry.js ./node_modules/@storybook/react/dist/esm/client/preview/config-generated-config-entry.js ./node_modules/@storybook/addon-links/dist/esm/preset/addDecorator.js-generated-config-entry.js ./.storybook/preview.js-generated-config-entry.js ./generated-stories-entry.js

ERROR in ./packages/cells/src/index.ts
Module not found: Error: Can't resolve '@glideapps/glide-data-grid' in '/Users/mike/dev/opensource/glide-data-grid/packages/cells/src'
 @ ./packages/cells/src/index.ts 1:0-58 15:22-34 26:22-34
 @ ./packages/cells/src/cell.stories.tsx
 @ . sync ^\.(?:(?:^|\/|(?:(?:(?!(?:^|\/)\.).)*?)\/)(?!\.)(?=.)[^/]*?\.stories\.tsx)$
 @ ./generated-stories-entry.js
 @ multi ./node_modules/@storybook/core-client/dist/esm/globals/polyfills.js ./node_modules/@storybook/core-client/dist/esm/globals/globals.js (webpack)-hot-middleware/client.js?reload=true&quiet=false&noInfo=undefined ./storybook-init-framework-entry.js ./node_modules/@storybook/react/dist/esm/client/preview/config-generated-config-entry.js ./node_modules/@storybook/addon-links/dist/esm/preset/addDecorator.js-generated-config-entry.js ./.storybook/preview.js-generated-config-entry.js ./generated-stories-entry.js

ERROR in ./packages/cells/src/cells/user-profile-cell.tsx
Module not found: Error: Can't resolve '@glideapps/glide-data-grid' in '/Users/mike/dev/opensource/glide-data-grid/packages/cells/src/cells'
 @ ./packages/cells/src/cells/user-profile-cell.tsx 2:0-78 32:20-37 62:31-44
 @ ./packages/cells/src/index.ts
 @ ./packages/cells/src/cell.stories.tsx
 @ . sync ^\.(?:(?:^|\/|(?:(?:(?!(?:^|\/)\.).)*?)\/)(?!\.)(?=.)[^/]*?\.stories\.tsx)$
 @ ./generated-stories-entry.js
 @ multi ./node_modules/@storybook/core-client/dist/esm/globals/polyfills.js ./node_modules/@storybook/core-client/dist/esm/globals/globals.js (webpack)-hot-middleware/client.js?reload=true&quiet=false&noInfo=undefined ./storybook-init-framework-entry.js ./node_modules/@storybook/react/dist/esm/client/preview/config-generated-config-entry.js ./node_modules/@storybook/addon-links/dist/esm/preset/addDecorator.js-generated-config-entry.js ./.storybook/preview.js-generated-config-entry.js ./generated-stories-entry.js
```

and then a bunch like this:
```
Child HtmlWebpackCompiler:
                          Asset      Size               Chunks  Chunk Names
    __child-HtmlWebpackPlugin_0  6.28 KiB  HtmlWebpackPlugin_0  HtmlWebpackPlugin_0
    Entrypoint HtmlWebpackPlugin_0 = __child-HtmlWebpackPlugin_0
    [./node_modules/html-webpack-plugin/lib/loader.js!./node_modules/@storybook/core-common/dist/cjs/templates/index.ejs] 2.03 KiB {HtmlWebpackPlugin_0} [built]
ModuleNotFoundError: Module not found: Error: Can't resolve '@glideapps/glide-data-grid' in '/Users/mike/dev/opensource/glide-data-grid/packages/cells/src'
    at /Users/mike/dev/opensource/glide-data-grid/node_modules/webpack/lib/Compilation.js:925:10
    at /Users/mike/dev/opensource/glide-data-grid/node_modules/webpack/lib/NormalModuleFactory.js:401:22
    at /Users/mike/dev/opensource/glide-data-grid/node_modules/webpack/lib/NormalModuleFactory.js:130:21
    at /Users/mike/dev/opensource/glide-data-grid/node_modules/webpack/lib/NormalModuleFactory.js:224:22
    at /Users/mike/dev/opensource/glide-data-grid/node_modules/neo-async/async.js:2830:7
    at /Users/mike/dev/opensource/glide-data-grid/node_modules/neo-async/async.js:6877:13
    at /Users/mike/dev/opensource/glide-data-grid/node_modules/webpack/lib/NormalModuleFactory.js:214:25
    at /Users/mike/dev/opensource/glide-data-grid/node_modules/enhanced-resolve/lib/Resolver.js:213:14
    at /Users/mike/dev/opensource/glide-data-grid/node_modules/enhanced-resolve/lib/Resolver.js:285:5
    at eval (eval at create (/Users/mike/dev/opensource/glide-data-grid/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at /Users/mike/dev/opensource/glide-data-grid/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:44:7
    at /Users/mike/dev/opensource/glide-data-grid/node_modules/enhanced-resolve/lib/Resolver.js:285:5
    at eval (eval at create (/Users/mike/dev/opensource/glide-data-grid/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at /Users/mike/dev/opensource/glide-data-grid/node_modules/enhanced-resolve/lib/Resolver.js:285:5
    at eval (eval at create (/Users/mike/dev/opensource/glide-data-grid/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
    at /Users/mike/dev/opensource/glide-data-grid/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:67:43
resolve '@glideapps/glide-data-grid' in '/Users/mike/dev/opensource/glide-data-grid/packages/cells/src'
  Parsed request is a module
  using description file: /Users/mike/dev/opensource/glide-data-grid/packages/cells/package.json (relative path: ./src)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      /Users/mike/dev/opensource/glide-data-grid/packages/cells/src/node_modules doesn't exist or is not a directory
      /Users/mike/dev/opensource/glide-data-grid/packages/cells/node_modules doesn't exist or is not a directory
      /Users/mike/dev/opensource/glide-data-grid/packages/node_modules doesn't exist or is not a directory
      /Users/mike/dev/opensource/node_modules doesn't exist or is not a directory
      /Users/mike/dev/node_modules doesn't exist or is not a directory
      /Users/mike/node_modules doesn't exist or is not a directory
      /Users/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /Users/mike/dev/opensource/glide-data-grid/node_modules
        using description file: /Users/mike/dev/opensource/glide-data-grid/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
          using description file: /Users/mike/dev/opensource/glide-data-grid/package.json (relative path: ./node_modules/@glideapps/glide-data-grid)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/mike/dev/opensource/glide-data-grid/node_modules/@glideapps/glide-data-grid doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/mike/dev/opensource/glide-data-grid/node_modules/@glideapps/glide-data-grid.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/mike/dev/opensource/glide-data-grid/node_modules/@glideapps/glide-data-grid.js doesn't exist
            .jsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/mike/dev/opensource/glide-data-grid/node_modules/@glideapps/glide-data-grid.jsx doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/mike/dev/opensource/glide-data-grid/node_modules/@glideapps/glide-data-grid.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/mike/dev/opensource/glide-data-grid/node_modules/@glideapps/glide-data-grid.tsx doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              /Users/mike/dev/opensource/glide-data-grid/node_modules/@glideapps/glide-data-grid.json doesn't exist
            .cjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/mike/dev/opensource/glide-data-grid/node_modules/@glideapps/glide-data-grid.cjs doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/mike/dev/opensource/glide-data-grid/node_modules/@glideapps/glide-data-grid.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/mike/dev/opensource/glide-data-grid/node_modules/@glideapps/glide-data-grid.tsx doesn't exist
            as directory
              /Users/mike/dev/opensource/glide-data-grid/node_modules/@glideapps/glide-data-grid doesn't exist

```

Let me know if any additional info would help.

**Comment by jassmith** [comment]
I see, you didn't run `npm run bootstrap` from the root if I had to guess.

**Comment by reczy** [comment]
Thanks, @jassmith - that was it. 

It wasn't clear that I needed to run that from `CONTRIBUTING.md` (I also needed to first have jq installed via `brew install jq`)

I was able to confirm that the bug is fixed in branch 4.0.0.

Cheers!

**Comment by jassmith** [comment]
Fixed contributing docs in 4.0
